src/model/WinPlayer.cs

```diff
@@ -11,116 +11,49 @@ namespace MusicSharp
-        private AudioFileReader audioFile;
+        private readonly WaveOutEvent outputDevice;
+        private AudioFileReader audioFileReader;
 
         /// <inheritdoc/>
         public string LastFileOpened { get; set; }
 
-        /// <inheritdoc/>
-        public WaveOutEvent OutputDevice
-        {
-            get
-            {
-                return this.outputDevice;
-            }
-
-            set
-            {
-                this.outputDevice = this.OutputDevice;
-            }
-        }
-
-        /// <inheritdoc/>
-        public AudioFileReader AudioFile
+        public WinPlayer() 
         {
-            get
-            {
-                return this.audioFile;
-            }
-
-            set
-            {
-                this.audioFile = this.AudioFile;
-            }
+            this.outputDevice = new WaveOutEvent();
+            this.outputDevice.PlaybackStopped += this.OnPlaybackStopped;
         }
 
         /// <inheritdoc/>
         public void Stop()
         {
-            if (this.outputDevice != null)
-            {
-                try
-                {
-                    this.outputDevice?.Stop();
-                    this.outputDevice.PlaybackStopped += this.OnPlaybackStopped;
-                }
-                catch (System.NullReferenceException)
-                {
-                }
-            }
+            this.outputDevice.Stop();
         }
 
         /// <inheritdoc/>
         public void Play(string path)
         {
-            if (this.outputDevice == null)
+            if (this.outputDevice.PlaybackState != PlaybackState.Stopped) 
             {
-                this.outputDevice = new WaveOutEvent();
-                this.outputDevice.PlaybackStopped += this.OnPlaybackStopped;
+                return;
             }
 
-            if (this.audioFile == null)
-            {
-                try
-                {
-                    this.audioFile = new AudioFileReader(this.LastFileOpened);
-                    this.outputDevice.Init(this.audioFile);
-                    this.outputDevice.Play();
-                }
-                catch (System.Runtime.InteropServices.COMException)
-                {
-                }
-            }
-
-            if (this.audioFile != null)
-            {
-                try
-                {
-                    this.outputDevice.Play();
-                }
-                catch (System.Runtime.InteropServices.COMException)
-                {
-                }
-            }
+            this.audioFileReader = new AudioFileReader(path);
+            this.outputDevice.Init(this.audioFileReader);
+            this.outputDevice.Play();
         }
 
         /// <inheritdoc/>
         public void Pause()
         {
-            try
-            {
-                this.outputDevice?.Pause();
-            }
-            catch (System.NullReferenceException)
-            {
-            }
+            this.outputDevice.Pause();
```

Comment:
Testing this PR, it seems that the pause button isn't working. After some review, it appears that lines 40-41 inside the Player method are creating a new instance of audioFileReader and outputDevice each time the Play button is pressed, which resets the pause method. 

My solution was to move the initialization inside an if statement inside the Play method like so:

public void Play(string path)
{
    if (this.outputDevice.PlaybackState == PlaybackState.Stopped)
    {
        this.audioFileReader = new AudioFileReader(path);
        this.outputDevice.Init(this.audioFileReader);
     }
    this.outputDevice.Play();
}